ecs/entity.go

```diff
@@ -8,7 +8,7 @@ import (
 // Systems, and has a number of Components
 type Entity struct {
 	id         string
-	components map[string]Component
+	components []Component
```

Comment:
Usually, it is. I forgot if it actually made A LOT of performance improvement, or just a little...